src/org/omegat/core/team2/RemoteRepositoryProvider.java

```diff
@@ -259,6 +259,28 @@ public void copyFilesFromReposToProject(String localPath) throws IOException {
         }
     }
 
+    /**
+     * Copy omegat.project from remote git server mapped to local
+     * directory under the name, omegat.project.NEW
+     *
+     * This is used only for that purpose, but follows the code of
+     * copyFilesFromReposToProject() because we have to go through the
+     * hoops of mapping to find out where the remote omegat.project
+     * file is.
+     *
+     * @param localPath
+     *            directory name or file name. This has to be "omegat.project".
+     */
+    public void copyFilesFromReposToProjectTempNew(String localPath) throws IOException {
+        String[] myForceExcludes = "".equals(localPath) ? forceExcludes : new String[]{};
+        for (Mapping m : getMappings(localPath, myForceExcludes)) {
+            m.copyFromRepoToProjectTempNew();
+            // if ("".equals(localPath)) {
+            //    m.propagateDeletes();
+            //}
```

Comment:
Please remove from line 278 to 280;
```
            // if ("".equals(localPath)) {
            //    m.propagateDeletes();
            //}
```

Reply:
Thank you reply and explanation.

> In my case I wanted to find the remote project file "omegat.project" for
the reason discussed in the ticket, but in the first try, I did not want
to disturb the original function (with the logic that handles the
"mapping".)
So I basically created the copy of the related functions and
modified them to accomplish what I wanted.

It is ok for your local work that you try to copy code and modify it to examine how it works, but,  IMO, it is not acceptable to merge into the project repository, and it should be formed to maintainable code.

For example,  we should define ` copyFilesFromReposToProject(final String localPath, final boolean propergateDelete)`.

```java
    /**
     * Copies all files under specified path that are mapped to local directory.
     * This is for backward comaptibility.
     */
    @Deprecated
    public void copyFilesFromReposToProject(String localPath) throws IOException {
        copyFilesFromReposToProject(localPath, true);
    }

    /**
     * Copies all files under specified path that are mapped to local directory.
     * <p>
     * If propagateDelete flag is set true and path is empty (i.e. full project),
     * also file deletions since last copy are propagated, i.e. if remote repo has
     * file deletions, those files are deleted locally as well.
     *
     * @param localPath
     *            directory name or file name
     */
    public void copyFilesFromReposToProject(final String localPath, final boolean propagateDelete) throws IOException {
        String[] myForceExcludes = "".equals(localPath) ? forceExcludes : new String[]{};
        for (Mapping m : getMappings(localPath, myForceExcludes)) {
            m.copyFromRepoToProject();
            if (propagateDelete && "".equals(localPath)) {
                m.propagateDeletes();
            }
        }
    }
```

Here is my suggested changes as diff form.

```diff
diff --git a/src/org/omegat/core/data/RealProject.java b/src/org/omegat/core/data/RealProject.java
index 2859750cc..05189af2f 100644
--- a/src/org/omegat/core/data/RealProject.java
+++ b/src/org/omegat/core/data/RealProject.java
@@ -345,7 +346,7 @@ public class RealProject implements IProject {
                     Log.logErrorRB("TEAM_NETWORK_ERROR", e.getCause());
                     setOfflineMode();
                 }
-                remoteRepositoryProvider.copyFilesFromReposToProject("");
+                remoteRepositoryProvider.copyFilesFromReposToProject("", true);
 
                 // After adding filters.xml and segmentation.conf, we must reload them again
                 config.loadProjectFilters();
diff --git a/src/org/omegat/core/team2/RemoteRepositoryProvider.java b/src/org/omegat/core/team2/RemoteRepositoryProvider.java
index cdc710c4d..dde065c5d 100644
--- a/src/org/omegat/core/team2/RemoteRepositoryProvider.java
+++ b/src/org/omegat/core/team2/RemoteRepositoryProvider.java
@@ -241,43 +241,33 @@ public class RemoteRepositoryProvider {
         }
     }
 
+
     /**
      * Copies all files under specified path that are mapped to local directory.
-     * If path is empty (i.e. full project), also file deletions since last copy are propagated, i.e. if remote repo has
-     * file deletions, those files are deleted locally as well.
-     *
-     * @param localPath
-     *            directory name or file name
+     * This is for backward comaptibility.
      */
+    @Deprecated
     public void copyFilesFromReposToProject(String localPath) throws IOException {
-        String[] myForceExcludes = "".equals(localPath) ? forceExcludes : new String[]{};
-        for (Mapping m : getMappings(localPath, myForceExcludes)) {
-            m.copyFromRepoToProject();
-            if ("".equals(localPath)) {
-                m.propagateDeletes();
-            }
-        }
+        copyFilesFromReposToProject(localPath, true);
     }
 
     /**
-     * Copy omegat.project from remote git server mapped to local
-     * directory under the name, omegat.project.NEW
-     *
-     * This is used only for that purpose, but follows the code of
-     * copyFilesFromReposToProject() because we have to go through the
-     * hoops of mapping to find out where the remote omegat.project
-     * file is.
+     * Copies all files under specified path that are mapped to local directory.
+     * <p>
+     * If propagateDelete flag is set true and path is empty (i.e. full project),
+     * also file deletions since last copy are propagated, i.e. if remote repo has
+     * file deletions, those files are deleted locally as well.
      *
      * @param localPath
-     *            directory name or file name. This has to be "omegat.project".
+     *            directory name or file name
      */
-    public void copyFilesFromReposToProjectTempNew(String localPath) throws IOException {
+    public void copyFilesFromReposToProject(final String localPath, final boolean propagateDelete) throws IOException {
         String[] myForceExcludes = "".equals(localPath) ? forceExcludes : new String[]{};
         for (Mapping m : getMappings(localPath, myForceExcludes)) {
-            m.copyFromRepoToProjectTempNew();
-            // if ("".equals(localPath)) {
-            //    m.propagateDeletes();
-            //}
+            m.copyFromRepoToProject();
+            if (propagateDelete && "".equals(localPath)) {
+                m.propagateDeletes();
+            }
         }
     }
 
diff --git a/src/org/omegat/gui/main/ProjectUICommands.java b/src/org/omegat/gui/main/ProjectUICommands.java
index 87dd01f96..3ede14fee 100644
--- a/src/org/omegat/gui/main/ProjectUICommands.java
+++ b/src/org/omegat/gui/main/ProjectUICommands.java
@@ -49,6 +49,7 @@ import javax.swing.SwingUtilities;
 import javax.swing.SwingWorker;
 
 import org.apache.commons.io.FileUtils;
+
 import org.omegat.CLIParameters;
 import org.omegat.convert.ConvertProject;
 import org.omegat.core.Core;
@@ -69,6 +70,7 @@ import org.omegat.gui.dialogs.NewTeamProject;
 import org.omegat.gui.dialogs.ProjectPropertiesDialog;
 import org.omegat.util.FileUtil;
 import org.omegat.util.FileUtil.ICollisionCallback;
+import org.omegat.util.HttpConnectionUtils;
 import org.omegat.util.Log;
 import org.omegat.util.OConsts;
 import org.omegat.util.OStrings;
@@ -77,7 +79,6 @@ import org.omegat.util.ProjectFileStorage;
 import org.omegat.util.RecentProjects;
 import org.omegat.util.StaticUtils;
 import org.omegat.util.StringUtil;
-import org.omegat.util.HttpConnectionUtils;
 import org.omegat.util.WikiGet;
 import org.omegat.util.gui.OmegaTFileChooser;
 import org.omegat.util.gui.OpenProjectFileChooser;
@@ -86,8 +87,6 @@ import org.omegat.util.gui.UIThreadsUtil;
 import gen.core.project.RepositoryDefinition;
 import gen.core.project.RepositoryMapping;
 
-import static org.omegat.util.ProjectFileStorage.loadPropertiesFile;
-
 /**
  * Handler for project UI commands, like open, save, compile, etc.
  *
@@ -346,7 +345,7 @@ public final class ProjectUICommands {
                 for (String file : new String[] { OConsts.FILE_PROJECT,
                         OConsts.DEFAULT_INTERNAL + '/' + FilterMaster.FILE_FILTERS,
                         OConsts.DEFAULT_INTERNAL + '/' + SRX.CONF_SENTSEG }) {
-                    remoteRepositoryProvider.copyFilesFromReposToProject(file);
+                    remoteRepositoryProvider.copyFilesFromReposToProject(file, true);
                 }
 
                 ProjectProperties props = ProjectFileStorage.loadProjectProperties(projectRoot);
@@ -494,7 +493,7 @@ public final class ProjectUICommands {
                                         new RemoteRepositoryProvider(props.getProjectRootDir(), props.getRepositories(), props);
                                 remoteRepositoryProvider.switchToVersion(OConsts.FILE_PROJECT, null);
                                 // Copy the content of remote omegat.project to top-level/omegat.project.NEW
-                                remoteRepositoryProvider.copyFilesFromReposToProjectTempNew(OConsts.FILE_PROJECT);
+                                remoteRepositoryProvider.copyFilesFromReposToProject(OConsts.FILE_PROJECT, false);
                                 rewriteOnSuccess = new File(projectRootFolder.getAbsoluteFile(), OConsts.FILE_PROJECT + ".NEW");
                                 // We should now load it from omegaT.project.NEW, instead.
                                 // We call loadPropertiesFile(... ) with "omegat.project.NEW".

```